### DISCUSSION TITLE
v0.5.6-alpha

### DISCUSSION THREAD

**Discussion by bowring** [CIRDLES/Tripoli#271] [Show and tell]
Ahoy!  We are back to developing on full steam.

This version 0.5.6 provides solutions to a variety of small issues raised recently in addition to bug fixes and UI improvements.
1.  Discussion #261, @YakovYermakov asks to be able to choose data columns before sculpting.  After consultation with Noah, the solution is that the plot window will not open the first time a specific Analysis Method is referred to in a data file.  As suggested by @ryanickert, the initial default will be all are selected.  Subsequently, the plot window will automatically open for any file using that Method.  On that first time, the user will select the columns to plot and then save them as the default for that method.  They can always be changed and saved later.  See image of new tab "Modify."  ![capture](https://github.com/user-attachments/assets/0df65db5-3ac9-4ac5-9ced-fb2fef39c97d)
2. Chauvenet's Criterion, discussed at length in various locations is now modified per @noahmclean's suggestions in Discussion #261: 
```text
Thinking about re-redefining some rules for Chauvenet:

You must have 20+ included points to use Chauvenet. This is because fewer points means a higher risk of rejecting non-outliers.
Chauvenet works on the set of all non-rejected data. If data have been rejected previously (e.g., a bad block tossed), then Chauvenet only considers the included data left over after that rejection. This is contrary to our current rules, but is a good change in my opinion.
After the Chauvenet rejection, the Chauvenet button grays out and becomes inactive so that you can't recursively apply Chauvenet.
After Chauvenet rejection, the user is still able to manually include/exclude data points they like/don't like -- even though they shouldn't.
After Chauvenet rejection, the user is able to undo Chauvenet by hitting the 'Restore Data' button. The 'Restore Data' button re-enables the Chauvenet button.
Thoughts from the peanut gallery?
```
3. Discussion #212, Noah raises the issue of hardcoded cycles per block for data files that are missing that info.  This version provides for modifying the cycles per block for ANY method.  See image above of new tab "Modify."  Note the "Reload Data" button that will re-parse the data per the now cycles per block.
4. General announcement - every data file that users have submitted to @noahmclean now opens automatically with drag'n'drop onto Tripoli.  If you have files that don't open, please send exemplars to @noahmclean.
5. Issue #264 - addressed here.  Make new issue if further refinements needed.
6. Issue #263 - fixed legend.
7. Issue #214 - @noahmclean's solution implemented - ratio will become user function in the presence of included negative or zero values. De-select them and user function returns to being a ratio.  v0.5.6

Cheers and please VOTE!

<hr /><em>This discussion was created from the release <a href='https://github.com/CIRDLES/Tripoli/releases/tag/v0.5.6-alpha'>v0.5.6-alpha</a>.</em>

**Comment by noahmclean** [comment]
This is great!!  

The "pick the time series to plot as defaults" option from Point 1 above is terrific.  However, every time you open a new file that is measured with one of the methods you have saved defaults for, the very first window of plots that pops up on opening contains all the time series.  You have to close it and then hit "Preview and Sculpt Data" to see the limited set.  Instead, for data files whose methods have defaults, the initial plots window should contain only the defaults.  

What a time-saver, though!

**Reply by bowring** [reply]
Fixed in v0.5.7; note that you don't need to close plots window to replot.
